core/src/com/unciv/logic/city/CityStats.kt

```diff
@@ -309,6 +311,20 @@ class CityStats {
                 stats.production += 15
         }
 
+        for (unique in cityInfo.cityConstructions.builtBuildingUniqueMap.getUniques("+[]% production when building [] in this city")) {
+            val filter = unique.params[1]
+            if (currentConstruction.name == filter
+                    || (filter == "land units" && currentConstruction is BaseUnit && currentConstruction.unitType.isLandUnit())
+                    || (filter == "naval units" && currentConstruction is BaseUnit && currentConstruction.unitType.isWaterUnit())
```

Comment:
"naval units" should appear in template.properties for translations if it's going to appear as a placeholder parameter